packages/data-context/src/data/ProjectConfigManager.ts

```diff
@@ -479,6 +488,7 @@ export class ProjectConfigManager {
       cliConfig: options.config ?? {},
       projectName: path.basename(this.options.projectRoot),
       projectRoot: this.options.projectRoot,
+      repoRoot: this.getRepoRoot(),
```

Comment:
Looks like this method is only called once - do we even need to define a `getRepoRoot`? Should we just do:

```js
repoRoot: this.options.ctx.git?.gitBaseDir
```

Then add the comment about `repoRoot` in `GitDataSource` or something?

Reply:
Ah yes, I originally had it as a one-liner, but felt for others navigating this code in the future it might be easy to miss. I added `getRepoRoot` to make it more explicit and have the long-form comment. Personally I think it's the best spot for it but if you or others feel strongly happy to move 😸